Example/SCNPath+Example/RealityKit Version/RKViewController+Gestures.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  RKViewController+Gestures.swift
+//  PathVisualiser
+//
+//  Created by Grant Jarvis on 03/13/21.
+//  Copyright © 2021 Grant Jarvis. All rights reserved.
+//
+
+
+import RealityKit
+import SceneKit
+
+
+extension RKViewController: UIGestureRecognizerDelegate {
+	func setupGestures() {
+		let tapGesture = UITapGestureRecognizer(target: self, action: #selector(handleTap(_:)))
+		tapGesture.delegate = self
+		self.view.addGestureRecognizer(tapGesture)
+	}
+
+	@IBAction func handleTap(_ gestureRecognizer: UITapGestureRecognizer) {
+		guard gestureRecognizer.state == .ended else {
+			return
+		}
+		if self.focusEntity.state != .initializing,
+           self.focusEntity.onPlane {
+            
+            self.hitPoints.append(self.focusEntity.position(relativeTo: nil).makeSCNVector3())
+		}
+	}
+}
+
+
+
+extension simd_float3 {
+    func makeSCNVector3() -> SCNVector3 {
+        return SCNVector3(self.x, self.y, self.z)
```

Comment:
This shouldn't be necessary, simd_float3 is defined as `typealias simd_float3 = SIMD3<Float>`, so they're interchangeable.

https://developer.apple.com/documentation/simd/simd_float3

Reply:
Yes, those two types are interchangeable but I believe that SCNVector3 is a third type from SceneKit that does need to be converted.

Reply:
oh you're right, I haven't poked around with SceneKit in a while I didn't even notice it's not SCNVector3, not SIMD3<Float>!
In that case, it can still be initialised with just SCNVector3(self) here

Reply:
which then means this method might not be needed, as it's just passing it directly to the initialiser

Reply:
Yes, good point. I also just realized that for the RealityKit implementation we can just get rid of SCNVector3 entirely if we make different arrays for self.hitPoints and pathNode.path that are Not arrays of SCNVector3.